homeassistant/auth.py

```diff
@@ -121,6 +317,77 @@ def async_create_credentials(self, data):
         return {}
 
 
+class LoginFlow(data_entry_flow.FlowHandler):
```

Comment:
This is the wrong architecture. 

I think that we should implement it as a FlowHandler that is not extended by the login flows, but instead wraps one.

It should take the login flow it needs to wrap. Once that flow returns a create entry, it should hold that value, look up the user, see if any two factor has been registered, instantiate those and let them handle the next step.

You can use `__getattr__` to intercept the methods on the LoginFlow and decide where to redirect the call.

Reply:
Hmm , maybe I do like this architecture. I just don't like how it iterates through the modules and mutates them.